starport/pkg/cosmosgen/generate_dart.go

```diff
@@ -0,0 +1,120 @@
+package cosmosgen
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/mattn/go-zglob"
+	"github.com/pkg/errors"
+	"github.com/tendermint/starport/starport/pkg/cosmosanalysis/module"
+	"github.com/tendermint/starport/starport/pkg/protoc"
+	protocgendart "github.com/tendermint/starport/starport/pkg/protoc-gen-dart"
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	dartOut = []string{
+		"--dart_out=grpc:.",
+	}
+)
+
+const (
+	dartExportFileName = "export.dart"
+	dartClientDirName  = "client"
+)
+
+type dartGenerator struct {
+	g *generator
+}
+
+func newDartGenerator(g *generator) *dartGenerator {
+	return &dartGenerator{
+		g: g,
+	}
+}
+
+func (g *generator) generateDart() error {
+	return newDartGenerator(g).generateModules()
+}
+
+func (g *dartGenerator) generateModules() error {
+	flag, cleanup, err := protocgendart.Flag()
+	if err != nil {
+		return err
+	}
+	defer cleanup()
+
+	gg := &errgroup.Group{}
+
+	add := func(sourcePath string, modules []module.Module) {
+		for _, m := range modules {
+			m := m
+			gg.Go(func() error { return g.generateModule(g.g.ctx, flag, sourcePath, m) })
+		}
+	}
+
+	add(g.g.appPath, g.g.appModules)
+
+	if g.g.o.dartIncludeThirdParty {
+		for sourcePath, modules := range g.g.thirdModules {
+			add(sourcePath, modules)
+		}
+	}
+
+	return gg.Wait()
+}
+
+func (g *dartGenerator) generateModule(ctx context.Context, plugin, appPath string, m module.Module) error {
+	var (
+		out       = g.g.o.dartOut(m)
+		clientOut = filepath.Join(out, dartClientDirName)
+		exportOut = filepath.Join(out, dartExportFileName)
+	)
+
+	includePaths, err := g.g.resolveInclude(appPath)
+	if err != nil {
+		return err
+	}
+
+	// reset destination dir.
+	if err := os.RemoveAll(out); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(clientOut, 0766); err != nil {
+		return err
+	}
+
+	// generate grpc client and protobuf types.
+	if err := protoc.Generate(
+		ctx,
+		clientOut,
+		m.Pkg.Path,
+		includePaths,
+		dartOut,
+		protoc.Plugin(plugin),
+		protoc.GenerateDependencies(),
+	); err != nil {
+		return err
+	}
+
+	// generate an export file to export all generated code through a single entrypoint.
+	generatedFiles, err := zglob.Glob(fmt.Sprintf("%s/**/*.dart", clientOut))
```

Comment:
can we use or improve the `localfs` package to fits here?

Reply:
```suggestion
	generatedFiles, err := zglob.Glob(filepath.Join(clientOut, "**/*.dart"))
```

Reply:
I don't think there is a need for it, this folder does not contain any hidden dirs.